tests/js/client/server_parameters/log-structured-params-json.js

```diff
@@ -0,0 +1,117 @@
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+////////////////////////////////////////////////////////////////////////////////
+
+const fs = require('fs');
+
+if (getOptions === true) {
+  return {
+    'log.structured-param': ['database=true', 'url', 'username', "pregelId=false"],
+    'log.use-json-format' : 'true',
+  };
+}
+
+const jsunity = require('jsunity');
+
+function LoggerSuite() {
+  'use strict';
+
+  return {
+
+    testStartingParameters: function () {
+      let res = arango.GET("/_admin/log/structured");
+      assertTrue(res.hasOwnProperty("database"));
+      assertTrue(res.hasOwnProperty("username"));
+      assertTrue(res.hasOwnProperty("url"));
+      assertFalse(res.hasOwnProperty("pregelId"));
+    },
+
+    testLogEntries: function() {
+      let res = arango.POST("/_admin/execute?returnBodyAsJSON=true", `
+require('console').log("testmann: start"); 
+for (let i = 0; i < 10; ++i) {
+  require('console').log("testmann: testi" + i);
+}
+require('console').log("testmann: done"); 
+return require('internal').options()["log.output"];
+`);
+
+      assertTrue(Array.isArray(res));
+      assertTrue(res.length > 0);
+
+      let logfile = res[res.length - 1].replace(/^file:\/\//, '');
+
+      // log is buffered, so give it a few tries until the log messages appear
+      let tries = 0;
+      let filtered = [];
+      while (++tries < 60) {
+        let content = fs.readFileSync(logfile, 'ascii');
+        let lines = content.split('\n');
+
+        filtered = lines.filter((line) => {
+          return line.match(/testmann: /);
+        });
+
+        if (filtered.length === 12) {
+          break;
+        }
+
+        require("internal").sleep(0.5);
+      }
+      assertEqual(12, filtered.length);
+
+      assertTrue(filtered[0].match(/testmann: start/));
+      for (let i = 1; i < 11; ++i) {
+        assertTrue(filtered[i].match(/testmann: testi\d+/));
+        assertTrue(filtered[i].match("[database: _system]"));
+        assertTrue(filtered[i].match("[username: root]"));
+        assertTrue(filtered[i].match(`[url: /_admin/execute?returnBodyAsJSON=true]`));
+      }
+      assertTrue(filtered[11].match(/testmann: done/));
+    },
+
+    testLogNewEntries: function() {
+      const res = arango.PUT("/_admin/log/structured", {"dog": true, "database": false});
+
+      assertTrue(res.hasOwnProperty("url"));
+      assertTrue(res.hasOwnProperty("username"));
+      assertFalse(res.hasOwnProperty("dog"));
+      assertFalse(res.hasOwnProperty("database"));
+    },
+    testLogRestoreParams: function() {
+      const res = arango.PUT("/_admin/log/structured", {"database": true});
+
+      assertTrue(Object.keys(res).length === 3);
+      assertTrue(res.hasOwnProperty("url"));
+      assertTrue(res.hasOwnProperty("username"));
+      assertFalse(res.hasOwnProperty("dog"));
+      assertTrue(res.hasOwnProperty("database"));
+    },
```

Comment:
These tests modify the global log settings and do not clean up after themselves. So the execution of tests in this file is fully order-dependent. This shouldn't be the case. If a test modifies global settings, it must clean up after itself.

In addition, these tests do not check whether the modified settings have any effect in newly created log messages.